diff

A patch to tweak Mozilla keyboard event handle for plugins so that
Komodo's SciMoz plugin works as we need. At least the second hunk here
(in `performKeyEquivalent:`) is to prevent Mozilla from special casing
plugins to NOT handle some keyboard events.

Index: widget/src/cocoa/nsChildView.mm
===================================================================
RCS file: /cvsroot/mozilla/widget/src/cocoa/nsChildView.mm,v
retrieving revision 1.362
diff -u -r1.362 nsChildView.mm
--- widget/src/cocoa/nsChildView.mm	22 Jul 2008 15:19:11 -0000	1.362
+++ widget/src/cocoa/nsChildView.mm	15 Aug 2008 00:56:52 -0000
@@ -5228,6 +5234,9 @@
 - (void)keyDown:(NSEvent*)theEvent
 {
   NS_OBJC_BEGIN_TRY_ABORT_BLOCK;
+/* XXX ACTIVESTATE: allowing TSM to be used on a plugin prevents keyPress
+                    events, thus breaking our use of SciMoz.
+                    
 
   // If a plugin has the focus, we need to use an alternate method for
   // handling NSKeyDown and NSKeyUp events (otherwise Carbon-based IME won't
@@ -5247,9 +5256,8 @@
     ::TSMRemoveDocumentProperty(mPluginTSMDoc, kFocusedChildViewTSMDocPropertyTag);
     return;
   }
-
+*/
   [self processKeyDownEvent:theEvent keyEquiv:NO];
-
   NS_OBJC_END_TRY_ABORT_BLOCK;
 }
 
@@ -5438,21 +5446,24 @@
                                [[self window] firstResponder] == self)) {
       return YES;
     }
   }
   else {  // Non-embedding
     // Perform native menu UI feedback even if we stop the event from propagating to it normally.
     // Recall that the menu system won't actually execute any commands for keyboard command invocations.
     //
     // If this is a plugin, we do actually perform the action on keyboard commands. See bug 428047.
     // If the action on plugins here changes the first responder, don't continue.
-    if (mIsPluginView) {
+    //
+    // ACTIVESTATE KOMODO: Always let Komodo's scimoz plugin handle the
+    // keyboard event (see Komodo bug 82895).
+    if (false && mIsPluginView) {
       [(GeckoNSMenu*)mainMenu actOnKeyEquivalent:theEvent];
       if ([[self window] firstResponder] != self)
         return YES;
     }
     else {
       [(GeckoNSMenu*)mainMenu performMenuUserInterfaceEffectsForEvent:theEvent];
     }
   }

   // With Cmd key or Ctrl+Tab or Ctrl+Esc, keyDown will be never called.
